pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -35,9 +37,9 @@ func ProvideService(
 			cache: make(map[int64]cachedDecrypted),
 		},
 	}
-	if remoteCheck.ShouldUseRemoteSecretsPlugin() {
+	if shouldUseRemoteSecretsPlugin(pluginsManager, cfg) {
```

Comment:
Based on the conversation in slack earlier, I think we can even do away with this logic and just have a check for the plugin being installed. This would support migrating secrets to and from the plugin without it necessarily being enabled. 

So something like:
```
if plugin installed
    start plugin

if use_plugin = true
   return wrapped remote plugin store
else 
   return wrapped sql store
        
```

Reply:
But what did we say about the scenario where the plugin is not installed but the `use_plugin = true` ? in that case the plugin would be nil if Im not wrong

Reply:
I believe `shouldUseRemoteSecretsPlugin` already accounts for `if plugin installed && use_plugin else sql_store` in this case this already works like we described on our conversations, I think migrating to the plugin without use_plugin enabled might not be a good approach since the user might not desire to have a bunch of secrets created on his secret store.